File: python/recommendations_client.py
import grpc

from recommendations_pb2 import BookCategory, RecommendationRequest
from recommendations_pb2_grpc import  RecommendationsStub


channel = grpc.insecure_channel("localhost:50051")
client = RecommendationsStub(channel)
request = RecommendationRequest(
    user_id = 1,
    category = BookCategory.SELF_HELP,
    max_results = 2
)

recommendations = client.Recommend(request)

print(recommendations)

File: recommendations/javascript/recommendations_client.js
// Imports
const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');

const packageDefinition = protoLoader.loadSync('../../protobufs/recommendations.proto', {});
const recommendationPackage = grpc.loadPackageDefinition(packageDefinition).recommendations;

const client = new recommendationPackage.Recommendations('localhost:50051', grpc.credentials.createInsecure());

client.Recommend(
  {
    'user_id': 12345,
    'category': recommendationPackage.BookCategory.SELF_HELP,
    'max_results': 5
  },
  (err, response) => {
    if (err) {
      console.log(err);
    } else {
      console.log('From server', JSON.stringify(response));
    }
  }
);
